File: python_backend/game_logic/threatArea/get_actual_pawn_threats.py
from game_logic.test_objects.sample_board_dicts import sample_board_dicts
from game_logic.sqr_case.get_sqr_case import get_sqr_case
from game_logic.piece_type.get_piece_type import get_piece_type
from game_logic.printers.print_board import print_board
from game_logic.sqr_case.sqr_cases import *


def get_actual_pawn_threats(board, potential_pawn_threats, color):
    """filter the list of potential pawn threats to get list of actual pawn threats"""
    return list(filter(lambda sqr: get_sqr_case(board, sqr[0], sqr[1], color) == ENEMY \
                                   and get_piece_type(board[sqr]) == 'P', potential_pawn_threats))


if __name__ == "__main__":
    board = sample_board_dicts['pawn_threat1']
    print('test 1')
    print_board(board, highlights=get_actual_pawn_threats(board, [(5, 4), (7, 4)], 'W'), highlights3=[(6, 3)])
    print('')
    print('test 2')
    print_board(board, highlights=get_actual_pawn_threats(board, [(2, 4), (4, 4)], 'B'), highlights3=[(3, 5)])
    print('')

File: node.js/threatArea/getActualPawnThreats.js
const  sampleBoardDicts  = require("../testObjects/sampleBoardDicts");
const  getSqrCase  = require("../sqrCase/getSqrCase");
const  getPieceType  = require("../pieceType/getPieceType");
const  printBoard  = require("../printers/printBoard");
const  {xyToRf} = require("../coordType/crdCnvrt");
const  {ENEMY}  = require("../sqrCase/sqrCases");
const mapListXyToRf = require("../coordType/mapListXyToRf");


function getActualPawnThreats(board, potentialPawnThreats, color) {
    /*filter the list of potential pawn threats to get list of actual pawn threats**/
    var potentialPawnThreats =  potentialPawnThreats.filter(sqr => getSqrCase(board, ...sqr, color) === ENEMY && getPieceType(board[xyToRf(...sqr)]) === 'P')
    return potentialPawnThreats;
}

module.exports = getActualPawnThreats;

if (require.main === module) {

    let board = sampleBoardDicts['pawn_threat1']

    printBoard(board, "\nwhite", mapListXyToRf(getActualPawnThreats(board, [[5, 4], [7, 4]], 'W'),  ["f3"]))
    printBoard(board, "\nblack", mapListXyToRf(getActualPawnThreats(board, [[2, 4], [4, 4]], 'B')), ["c5"])
}